k8s/maintenance/phpmyadmin.yaml

```diff
@@ -0,0 +1,44 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: phpmyadmin
+  labels:
+    app: phpmyadmin
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app: phpmyadmin
+  template:
+    metadata:
+      labels:
+        app: phpmyadmin
+    spec:
+      containers:
+        - name: phpmyadmin
+          image: phpmyadmin:5.1-apache
+          ports:
+            - containerPort: 80
+          env:
+            - name: PMA_HOST
+              value: mariadb
+            - name: PMA_PORT
+              value: "3306"
+            - name: MYSQL_ROOT_PASSWORD
+              valueFrom:
+                secretKeyRef:
+                  name: mdbsecrets
+                  key: password
+---
+apiVersion: v1
+kind: Service
+metadata:
+  name: phpmyadmin
+spec:
+  type: NodePort
```

Comment:
Probably not a `NodePort` service. See https://kubernetes.io/docs/concepts/services-networking/service/#publishing-services-service-types. Basicallly a `NodePort` requires that the port be globally unique across the cluster and would require a lot of coordination to get right. `NodePort` is generally used as a "last resort" to expose something that can't be exposed any other way (e.g. our Ambassador service is `NodePort` so we can give Continuu a port number to which they can send traffic to route it into our cluster). The default `ClusterIP` is fine here as Ambassador can route to that and you can also reach it via `kubectl port-forward`.

Reply:
Will get that to `ClusterIP`. Thanks for the explanation and the link. Learning little by little.